include/boost/thread/detail/thread.hpp

```diff
@@ -539,17 +529,14 @@ namespace thread_detail {
 #if defined BOOST_THREAD_USES_DATETIME
         bool timed_join(const system_time& abs_time)
         {
-          struct timespec const ts=detail::to_timespec(abs_time);
-          return do_try_join_until(ts);
+          return do_try_join_until(detail::timespec_to_internal_clock(abs_time));
         }
 #endif
 #ifdef BOOST_THREAD_USES_CHRONO
-        bool try_join_until(const chrono::time_point<thread_detail::internal_clock_t, chrono::nanoseconds>& tp)
+        template <class Duration>
+        bool try_join_until(const chrono::time_point<thread_detail::internal_clock_t, Duration>& tp)
         {
-          using namespace chrono;
-          nanoseconds d = tp.time_since_epoch();
-          timespec ts = boost::detail::to_timespec(d);
-          return do_try_join_until(ts);
+          return do_try_join_until(boost::detail::to_timespec(tp.time_since_epoch()));
```

Comment:
This has a sense only on Pthread. Windows has no timespec.

Reply:
Agreed. This particular instance is in an #else section that's only enabled when BOOST_THREAD_PLATFORM_WIN32 is not defined.

Reply:
RIght :)